Content/Items/Breacher/Weapons.FlareBreacher.cs

```diff
@@ -350,4 +441,165 @@ public void DrawPrimitives()
 			trail?.Render(effect);
 		}
 	}
+
+	public class FlareBreacherMuzzleFlashDust : ModDust
+	{
+		public override string Texture => AssetDirectory.Invisible;
+
+		public override void OnSpawn(Dust dust)
+		{
+			dust.frame = new Rectangle(0, 0, 4, 4);
+		}
+
+		public override bool Update(Dust dust)
+		{
+			dust.alpha += 15;
+			dust.alpha = (int)(dust.alpha * 1.05f);
+
+			if (dust.alpha >= 255)
+				dust.active = false;
+
+			return false;
+		}
+
+		public override bool PreDraw(Dust dust)
+		{
+			float lerper = 1f - dust.alpha / 255f;
+
+			Texture2D tex = ModContent.Request<Texture2D>(AssetDirectory.BreacherItem  + Name).Value;
```

Comment:
This all needs to be changed to the AssGen assets, this will cause severe memory usage especially on a dust